pexpect/pty_spawn.py

```diff
@@ -210,7 +210,8 @@ def __str__(self):
         s.append('match: %r' % (self.match,))
         s.append('match_index: ' + str(self.match_index))
         s.append('exitstatus: ' + str(self.exitstatus))
-        s.append('flag_eof: ' + str(self.flag_eof))
+        if hasattr(self, 'ptyproc'):
+            s.append('flag_eof: ' + str(self.flag_eof))
```

Comment:
This feels distinctly band-aid-ish. I'm happy for it to go in so that other errors show up as expected, but we should probably look more closely at whether `self.flag_eof` can do something sensible in such subclasses.

Even thinking about thinking about that makes me sleepy ;-)